src/shared/bpf-link.c

```diff
@@ -5,15 +5,33 @@
 #endif
 
 #include "bpf-link.h"
+#include "serialize.h"
 
 #if HAVE_LIBBPF
+int serialize_bpf_link(FILE *f, FDSet *fds, const char *key, BPFLink *link) {
+        int fd;
+
+        assert(key);
+        assert(link);
+
+        fd = bpf_link__fd(link);
+        if (fd < 0)
+                return -errno;
```

Comment:
doesn't `bpf_link__fd` return error codes as function return params? i.e. shouldn#t this do `return fd` here?

Reply:
this will never return fd < 0, so this check is not necessary.